### PR TITLE
branch to investigate problems in AdaptorNeighborlist

### PR DESCRIPTION
**Pull request by felixmusil**
We get an segfault in AdaptorNeighborlist with a seemingly sane structure.

The error is reproduced in playground.cc if filename == "reference_data/alloy.json" and Michele made it work in "reference_data/alloy-fixed.json" by wrapping and shifting the atoms by the center of mass:
```
frame = read("../tests/reference_data/alloy.json","0")
frame.wrap(center=(0., 0., 0.))
write("../tests/reference_data/alloy-fixed.json",frame)
```

Basically with  "reference_data/alloy.json" when binning the j-atoms in the boxes some atoms end up outside of the bounds leading to a segfault.

It seems from Michele's investigations that alloy-fixed.json is still interpreted properly since the soap vectors are very (too) different from the ones of quippy.

### CODE REVIEW

**Comment by max-veit** [comment]
see also #126 for problems with "insane" structures (i.e. the ones that violate the -- still unwritten -- assumptions behind this linked-cell algorithm)

**Comment by ceriottm** [comment]
yeah, I know that @max-veit raised a red flag already quite some time ago - I kind of thought it was a corner-case issue, so I thought it was not priority, but given that it seems this crashes (or worse, gives wrong results) when one passes a simple replicated FCC lattice, I think it needs maximum attention and a quick, robust fix - to be followed up with less urgency with a more efficient implementation, possibly relying on sanitizing the simulation box and/or folding the atoms

**Comment by mastricker** [comment]
I'll have a go at it once I'm back on Friday.

My understanding was the same: corner case an we leave it for the future. Seems more pressing now.

This branch has everything that is needed to replicate the issue?

On Sep 29, 2019 17:36, Michele Ceriotti <notifications@github.com> wrote:

yeah, I know that @max-veit<https://github.com/max-veit> raised a red flag already quite some time ago - I kind of thought it was a corner-case issue, so I thought it was not priority, but given that it seems this crashes (or worse, gives wrong results) when one passes a simple replicated FCC lattice, I think it needs maximum attention and a quick, robust fix - to be followed up with less urgency with a more efficient implementation, possibly relying on sanitizing the simulation box and/or folding the atoms

-

File: examples/linked_cells.ipynb [hunk 1]
```diff
@@ -68,38 +62,42 @@
...
   {
    "cell_type": "code",
-   "execution_count": null,
-   "metadata": {
-    "hidden": true
-   },
-   "outputs": [],
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [
```
Comment (max-veit) [inline examples/linked_cells.ipynb]: errrrghhh, please install the notebook cleaner hook, we really don't need another series of outputs recorded in history forever

**Comment by mastricker** [comment]
Is this solved now?

On Oct 3, 2019 20:42, felixmusil <notifications@github.com> wrote:

Closed #132<https://github.com/cosmo-epfl/librascal/pull/132>.

-

**Pull request closed without merging**